src/crypto/tls/common.go

```diff
@@ -350,6 +350,13 @@ type ConnectionState struct {
 	// This feature is unstable and applications MUST NOT depend on it.
 	CFControl interface{}
 
+	// CFClientHello is the serialized ClientHello used to terminate the
+	// connection.
```

Comment:
It might be worth pointing out that this is only set when the handshake has successfully been completed.
"terminate the connection" might be confusing who are not familiar with ECH. I suggest changing it, here is a proposal based on the existing documentation for "Version" and "HandshakeComplete".

```suggestion
	// CFClientHello is the serialized ClientHello handshake message used by the
	// connection. It can be empty if the handshake has not concluded.
```